monitoring/monitorlib/scd.py

```diff
@@ -167,16 +99,203 @@ class Volume3D(ImplicitDict):
-################################################################################
\ No newline at end of file
+################################################################################
+
+
+def make_vol4(
+        t0: Optional[datetime] = None,
+        t1: Optional[datetime] = None,
+        alt0: Optional[float] = None,
+        alt1: Optional[float] = None,
+        circle: Dict = None,
+        polygon: Dict = None) -> Volume4D:
+    kwargs = dict()
+    if circle is not None:
+        kwargs['outline_circle'] = circle
+    if polygon is not None:
+        kwargs['outline_polygon'] = polygon
+    if alt0 is not None:
+        kwargs['altitude_lower'] = make_altitude(alt0)
+    if alt1 is not None:
+        kwargs['altitude_upper'] = make_altitude(alt1)
+    vol3 = Volume3D(**kwargs)
+    kwargs = {'volume': vol3}
+    if t0 is not None:
+        kwargs['time_start'] = make_time(t0)
+    if t1 is not None:
+        kwargs['time_end'] = make_time(t1)
+    return Volume4D(**kwargs)
+
+
+def make_time(t: datetime) -> Time:
+    return Time(value=t.isoformat() + 'Z', format='RFC3339')
+
+
+def make_altitude(alt_meters: float) -> Altitude:
+    return Altitude(value=alt_meters, reference='W84', units='M')
+
+
+def make_circle(lat: float, lng: float, radius: float) -> Circle:
+    return Circle(
+        center=LatLngPoint(lat=lat, lng=lng),
+        radius=Radius(value=radius, units='M'))
+
+
+def make_polygon(coords: List[Tuple[float, float]]=None, latlngrect: s2sphere.LatLngRect=None) -> Polygon:
+    if coords is not None:
+        return Polygon(vertices=[LatLngPoint(lat=lat, lng=lng) for (lat, lng) in coords])
+
+    return Polygon(vertices=[
+        LatLngPoint(lat=latlngrect.lat_lo().degrees, lng=latlngrect.lng_lo().degrees),
+        LatLngPoint(lat=latlngrect.lat_lo().degrees, lng=latlngrect.lng_hi().degrees),
+        LatLngPoint(lat=latlngrect.lat_hi().degrees, lng=latlngrect.lng_hi().degrees),
+        LatLngPoint(lat=latlngrect.lat_hi().degrees, lng=latlngrect.lng_lo().degrees),
+    ])
+
+
+def start_of(vol4s: List[Volume4D]) -> datetime:
+    return min([parse_time(vol4['time_start']) for vol4 in vol4s])
+
+
+def end_of(vol4s: List[Volume4D]) -> datetime:
+    return max([parse_time(vol4['time_end']) for vol4 in vol4s])
+
+
+def rect_bounds_of(vol4s: List[Volume4D]) -> s2sphere.LatLngRect:
+    lat_min = 90
+    lat_max = -90
+    lng_min = 360
+    lng_max = -360
+    for vol4 in vol4s:
+        if 'outline_polygon' in vol4.volume:
+            for v in vol4.volume.outline_polygon.vertices:
+                lat_min = min(lat_min, v.lat)
+                lat_max = max(lat_max, v.lat)
+                lng_min = min(lng_min, v.lng)
+                lng_max = max(lng_max, v.lng)
+        if 'outline_circle' in vol4.volume:
+            circle = vol4.volume.outline_circle
+            lat_radius = 360 * circle.radius.value / geo.EARTH_CIRCUMFERENCE_M
```

Comment:
Just noticed that only Meter is accepted by [specification](https://github.com/astm-utm/Protocol/blob/f6e5ee5a84cbe56d1b39187aa5ba0f67a7d8f5ec/utm.yaml#L186).

Reply:
Yes, only meters for now.  It does make me a little uneasy that this will break if/when the API changes to allow other units, but a lot of stuff will break due to a changing API and I don't think this is unique enough to warrant a full-blown generalized units handler.  It doesn't hurt to be more defensive though, so I've now added checks that throw exceptions when units aren't M.